internal/driver/driver.go

```diff
@@ -2,354 +2,151 @@
 //
 // Copyright (C) 2018 Canonical Ltd
 // Copyright (C) 2018 IOTech Ltd
+// Copyright (C) 2021 Schneider Electric
 //
 // SPDX-License-Identifier: Apache-2.0
 
-//
 package driver
 
 import (
 	"context"
 	"fmt"
-	"log"
 	"sync"
-	"time"
 
-	"github.com/edgexfoundry/device-sdk-go"
-	sdkModel "github.com/edgexfoundry/device-sdk-go/pkg/models"
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
-	"github.com/edgexfoundry/go-mod-core-contracts/models"
-	"github.com/gopcua/opcua"
-	"github.com/gopcua/opcua/ua"
-	"github.com/spf13/cast"
+	"github.com/edgexfoundry/device-opcua-go/internal/config"
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	"github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
 )
 
 var once sync.Once
 var driver *Driver
 
+// Driver struct
 type Driver struct {
 	Logger           logger.LoggingClient
 	AsyncCh          chan<- *sdkModel.AsyncValues
 	CommandResponses sync.Map
-	Config           *configuration
+	serviceConfig    *config.ServiceConfig
+	resourceMap      map[uint32]string
```

Comment:
to prevent the race condition issue, lock should be added when accessing this `resourceMap`

Reply:
Please let me know if you think https://github.com/edgexfoundry-holding/device-opcua-go/pull/9/commits/a76b19d0eada235c16494282ef1bfc4cadcedf8e satisfies this request. Thanks!

Reply:
Updated to deal with the loops better in https://github.com/edgexfoundry-holding/device-opcua-go/pull/9/commits/83ad36c5b971654aa8b8b78357ffc5a3ee254203